snap/component.go

```diff
@@ -143,6 +146,41 @@ func (c *componentPlaceInfo) MountDescription() string {
 	return fmt.Sprintf("Mount unit for %s, revision %s", c.ContainerName(), c.compRevision)
 }
 
+// ComponentLinkPath returns the path for the symlink for a component for a
+// given snap revision.
+func ComponentLinkPath(cpi ContainerPlaceInfo, snapRev Revision) string {
+	instanceName, compName, _ := strings.Cut(cpi.ContainerName(), "+")
+	compBase := ComponentsBaseDir(instanceName)
+	return filepath.Join(compBase, snapRev.String(), compName)
+}
+
+// ComponentInstallDate returns the "install date" of the component by checking
+// when its symlink was created. We cannot use the mount directory as lstat
+// returns the date of the root of the container instead of the date when the
+// mount directory was created.
+func ComponentInstallDate(cpi ContainerPlaceInfo, snapRev Revision) *time.Time {
+	symLn := ComponentLinkPath(cpi, snapRev)
+	if st, err := os.Lstat(symLn); err == nil {
+		modTime := st.ModTime()
+		return &modTime
+	}
+	return nil
+}
+
+// ComponentSize returns the file size of a component.
+func ComponentSize(cpi ContainerPlaceInfo) int64 {
```

Comment:
This also feels like it should return an error here if something goes wrong, and make the caller decide if it wants to default to zero?

Reply:
Changed now